Tests/Tests/STPBankAccountTest.m

```diff
@@ -100,53 +117,82 @@ - (void)testBankAccountEquals {
-    STPBankAccount *bankAccountWithAttributes = [STPBankAccount decodedObjectFromAPIResponse:apiResponse];
-
-    XCTAssertEqualObjects([bankAccountWithAttributes bankAccountId], @"something", @"bankAccountId is set correctly");
-    XCTAssertEqualObjects([bankAccountWithAttributes last4], @"6789", @"last4 is set correctly");
-    XCTAssertEqualObjects([bankAccountWithAttributes bankName], @"STRIPE TEST BANK", @"bankName is set correctly");
-    XCTAssertEqualObjects([bankAccountWithAttributes country], @"US", @"country is set correctly");
-    XCTAssertEqualObjects([bankAccountWithAttributes fingerprint], @"something", @"fingerprint is set correctly");
-    XCTAssertEqualObjects([bankAccountWithAttributes currency], @"usd", @"currency is set correctly");
-    XCTAssertEqual([bankAccountWithAttributes status], STPBankAccountStatusNew);
-    XCTAssertEqualObjects([bankAccountWithAttributes accountHolderName], @"John Doe");
-    XCTAssertEqual([bankAccountWithAttributes accountHolderType], STPBankAccountHolderTypeCompany);
-
-    NSDictionary *allResponseFields = bankAccountWithAttributes.allResponseFields;
-    XCTAssertEqual(allResponseFields[@"foo"], @"bar");
-    XCTAssertEqual(allResponseFields[@"last4"], @"6789");
-    XCTAssertNil(allResponseFields[@"baz"]);
+- (void)testDecodedObjectFromAPIResponseRequiredFields {
+    NSArray<NSString *> *requiredFields = @[
+                                            @"id",
+                                            @"last4",
+                                            @"bank_name",
+                                            @"country",
+                                            @"currency",
+                                            @"status",
+                                            ];
+
+    for (NSString *field in requiredFields) {
+        NSMutableDictionary *response = [[self completeAttributeDictionary] mutableCopy];
+        [response removeObjectForKey:field];
+
+        XCTAssertNil([STPBankAccount decodedObjectFromAPIResponse:response]);
+    }
+
+    XCTAssert([STPBankAccount decodedObjectFromAPIResponse:[self completeAttributeDictionary]]);
+}
+
+- (void)testDecodedObjectFromAPIResponseMapping {
+    NSDictionary *response = [self completeAttributeDictionary];
+    STPBankAccount *bankAccount = [STPBankAccount decodedObjectFromAPIResponse:response];
+
+    XCTAssertEqualObjects(bankAccount.bankAccountId, @"ba_1AXvnKEOD54MuFwSotKc6xq0");
+    XCTAssertEqualObjects(bankAccount.accountHolderName, @"Jane Austen");
+    XCTAssertEqual(bankAccount.accountHolderType, STPBankAccountHolderTypeIndividual);
+    XCTAssertEqualObjects(bankAccount.bankName, @"STRIPE TEST BANK");
+    XCTAssertEqualObjects(bankAccount.country, @"US");
+    XCTAssertEqualObjects(bankAccount.currency, @"usd");
+    XCTAssertEqualObjects(bankAccount.fingerprint, @"C5fW7AwE3of8bHvV");
+    XCTAssertEqualObjects(bankAccount.last4, @"6789");
+    XCTAssertNil(bankAccount.routingNumber);
```

Comment:
Cool, I'll at least map the `routingNumber`. I think overall `STPBankAccount` has a few nullability bugs that we can fix (cause public api changes) but I'm guessing the object isn't used very often so we don't hear much about it.

Are we ever using the fact that `STPBankAccount` is a subclass of `STPBankAccountParams` to allow people to retrieve/modify/resubmit a bank account easily? Maybe that's too magical.